File: src/main/java/net/hectus/neobb/turn/person_game/warp/PTVillagerWarp.java
package net.hectus.neobb.turn.person_game.warp;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.util.Modifiers;
import org.bukkit.World;

public class PTVillagerWarp extends PWarpTurn {
    public PTVillagerWarp(World world) { super(world, "villager"); }
    public PTVillagerWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "villager"); }

    @Override
    public void apply() {
        super.apply();
        player.game.addModifier(Modifiers.G_PERSON_VILLAGER_ENTITIES);
        player.inventory.addRandom();
    }

    @Override
    public int chance() {
        return 40;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/person_game/warp/PTVillagerWarp.kt
package net.hectus.neobb.turn.person_game.warp

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.structure.StaticStructure
import net.hectus.neobb.structure.StaticStructures
import net.hectus.neobb.util.Modifiers

class PTVillagerWarp(data: PlacedStructure?, cord: Cord, player: NeoPlayer?) : PWarpTurn(data, cord, "villager", player) {
    override val staticStructure: StaticStructure = StaticStructures.Person.Warp.VILLAGER

    override val chance: Double = 40.0

    override fun apply() {
        player!!.game.addModifier(Modifiers.Game.Person.VILLAGER_ENTITIES)
        player.inventory.addRandom()
    }
}
